File: TS/1. Two Sum.ts
function twoSum(nums: number[], target: number): number[] {
  const hashMap: { [key: number]: number } = {};

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] in hashMap) {
      return [hashMap[nums[i]], i];
    }

    hashMap[target - nums[i]] = i;
  }

  return [];
}

console.log(twoSum([2, 7, 11, 15], 9));
console.log(twoSum([3, 2, 4], 6));

File: Python/1. Two Sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap={}
        for i in range(len(nums)):
            if nums[i] in hashMap:
                return [hashMap[nums[i]], i]

            hashMap[target-nums[i]] = i
        return []


if __name__ == "__main__":
    solution = Solution()
    print(solution.twoSum([2, 7, 11, 15], 9))
    print(solution.twoSum([3, 2, 4], 6))